simulator/src/resim/mod.rs

```diff
@@ -304,16 +306,237 @@ pub fn export_abi(
     blueprint_name: &str,
 ) -> Result<abi::BlueprintAbi, Error> {
     let scrypto_interpreter = ScryptoInterpreter::<DefaultWasmEngine>::default();
-    let mut substate_store = RadixEngineDB::with_bootstrap(get_data_dir()?, &scrypto_interpreter);
-    radix_engine::model::export_abi(&mut substate_store, package_address, blueprint_name)
+    let substate_store = RadixEngineDB::with_bootstrap(get_data_dir()?, &scrypto_interpreter);
+    radix_engine::model::export_abi(&substate_store, package_address, blueprint_name)
         .map_err(Error::AbiExportError)
 }
 
 pub fn export_abi_by_component(
     component_address: ComponentAddress,
 ) -> Result<abi::BlueprintAbi, Error> {
-    let scrypto_interpreter = ScryptoInterpreter::<DefaultWasmEngine>::default();
-    let mut substate_store = RadixEngineDB::with_bootstrap(get_data_dir()?, &scrypto_interpreter);
-    radix_engine::model::export_abi_by_component(&mut substate_store, component_address)
-        .map_err(Error::AbiExportError)
+    match component_address {
+        ComponentAddress::Account(..)
+        | ComponentAddress::EcdsaSecp256k1VirtualAccount(..)
+        | ComponentAddress::EddsaEd25519VirtualAccount(..) => Ok(export_account_abi()),
+        ComponentAddress::Normal(..) => {
+            let scrypto_interpreter = ScryptoInterpreter::<DefaultWasmEngine>::default();
+            let substate_store =
+                RadixEngineDB::with_bootstrap(get_data_dir()?, &scrypto_interpreter);
+            radix_engine::model::export_abi_by_component(&substate_store, component_address)
+                .map_err(Error::AbiExportError)
+        }
+
+        _ => todo!("Unsupported native ABI."),
+    }
+}
+
+// Keeping this function here for the time being as it helps me make stronger assumptions about the
+// context in which it will be used.
+// TODO: Refactor out once we have a clear plan with native component package ABIs
+pub fn export_account_abi() -> abi::BlueprintAbi {
```

Comment:
Would appreciate input on this. How do we plan to deal with package ABIs with native packages?

Reply:
We're moving towards "everything has an ABI including native packages"

Reply:
Makes sense. I'll keep this here until we have native package ABI implemented.